finance-tracker/app/models/user.rb

```diff
@@ -5,4 +5,21 @@ class User < ApplicationRecord
   # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable
+
+  def stock_already_tracked?(ticker_symbol)
+    stocks.exists?(ticker: ticker_symbol)
+  end
+
+  def under_stock_limit?
+    stocks.count < 10
+  end
+
+  def can_track_stock?(ticker_symbol)
+    under_stock_limit? && !stock_already_tracked?(ticker_symbol)
+  end
+
+  def full_name
+    return "#{first_name} #{last_name}" if first_name || last_name
+    "Anonymous"
```

Comment:
```suggestion
    (first_name || last_name) ? "#{first_name} #{last_name}" : "Anonymous"
```
might be easier to read. wdyt?